### PR TITLE
Farmbot tweak/fix

### LINKED ISSUE #5718: Farmbots still aren't fixed.
Despite numerous attempts to fix, they still aren't filling themselves at the hydroponics sink.

### LINKED ISSUE #5721: Basketball court on odin has misalligned windoors
court

### PR DESCRIPTION
**Pull request by Sindorman**
- Farmbots prioritize refilling if their tanks are below 30%

- Farmbots option to turn On/Off watering trays actually works, was set to use wrong var.

- Moved sink in Hydroponics so that it is actually is in field of view of farmbots when they attend to trays. Fixes #5718 

- Reworded APC and SMES state when input and output are the same

- Fixes #5721

### CODE REVIEW

File: code/modules/mob/living/bot/farmbot.dm [hunk 1]
```diff
@@ -163,7 +163,7 @@
 					target = tray
 					frustration = 0
 					break
-			if(!target && refills_water && tank && tank.reagents.total_volume < tank.reagents.maximum_volume)
+			if((!target && refills_water && tank && tank.reagents.total_volume < tank.reagents.maximum_volume) || ((tank.reagents.maximum_volume - tank.reagents.total_volume ) / reagents.maximum_volume) <= 0.3)
```
Comment (skull132) [inline code/modules/mob/living/bot/farmbot.dm]: Make this check into a function. It's way too long.

File: code/modules/mob/living/bot/farmbot.dm [hunk 2]
```diff
@@ -348,7 +348,7 @@
 		user << "You complete the Farmbot! Beep boop."
 		var/mob/living/bot/farmbot/S = new /mob/living/bot/farmbot(get_turf(src))
 		for(var/obj/structure/reagent_dispensers/watertank/wTank in contents)
-			qdel(S.tank)
+			QDEL_NULL(S.tank)
```
Comment (skull132) [inline code/modules/mob/living/bot/farmbot.dm]: Why is this necessary? The loop sets a new reference already.

Comment (Sindorman) [reply code/modules/mob/living/bot/farmbot.dm]: *shrug emoji* so that it deletes before setting a new reference.
